.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: test
+
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version:
+          - "3.12"
+          - "3.11"
+          - "3.10"
+          - "3.9"
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          pip install poetry
+          poetry install
```

Comment:
Could you get away with just installing dev dependencies?

Reply:
It needs the others too for the production code to run.